scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+import numpy as np
+import math
+from . import linear_sum_assignment, minimize_scalar, OptimizeResult
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    seed_cost=[],
+    seed_dist=[],
+    maximize=False,
+    n_init=1,
+    init_method="barycenter",
+    max_iter=30,
+    shuffle_input=True,
+    eps=0.05,
+):
+    """
+    Solve the quadratic assignment problem
+
+    This class solves the Quadratic Assignment Problem and the Graph
+    Matching Problem (QAP) through an implementation of the Fast
+    Approximate QAP Algorithm (FAQ) (these two problems are the same up
+    to a sign change) [1].
+
+    This algorithm can be thought of as finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2].
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square, positive
+        A square adjacency matrix
+
+    seed_cost : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'cost_matrix'
+        (shape (m , 1) where m <= number of nodes).
+
+    seeds_dist : 1d-array, optional, (default = [])
+        An array where each entry is an index of a node in 'dist_matrix'
+        The elements of 'seed_cost' and 'seed_dist' are vertices which
+        are known to be matched, that is, 'seed_cost[i]' is matched to
+        vertex 'seed_dist[i]'. Array shape (m , 1)
+        where m <= number of nodes
+
+    maximize : bool (default = False)
+        Gives users the option to solve the Graph Matching Problem (GMP)
+        rather than QAP. This is accomplished through trivial negation
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+               Number of random initializations of the starting
+               permutation matrix that the FAQ algorithm will undergo.
+               n_init automatically set to 1 if init_method = 'barycenter'
+            init_method : string (default = 'barycenter')
+               The initial position chosen:
```

Comment:
Is it possible to very briefly note what influence this "position" has on the initialization of the starting permutation matrix?